### PR TITLE
replace deprecated confluence group api endpoint

### PR DESCRIPTION
**Pull request by hagen-danswer**
## Description
^

## Backporting (check the box to trigger backport action)
Note: You have to check that the action passes, otherwise resolve the conflicts manually and tag the patches.
- [ ] This PR should be backported (make sure to check that the backport attempt succeeds)

### CODE REVIEW

File: backend/danswer/connectors/confluence/utils.py [hunk 1]
```diff
@@ -269,20 +269,3 @@ def datetime_from_string(datetime_string: str) -> datetime:
         datetime_object = datetime_object.astimezone(timezone.utc)
 
     return datetime_object
-
-
```
Comment (hagen-danswer) [inline backend/danswer/connectors/confluence/utils.py:273]: moved this to the onyx_confluence.py file

File: backend/ee/danswer/external_permissions/confluence/group_sync.py [hunk 2]
```diff
@@ -38,41 +62,21 @@ def confluence_group_sync(
     is_cloud = cc_pair.connector.connector_specific_config.get("is_cloud", False)
     wiki_base = cc_pair.connector.connector_specific_config["wiki_base"]
 
-    # test connection with direct client, no retries
```
Comment (hagen-danswer) [inline backend/ee/danswer/external_permissions/confluence/group_sync.py:41]: moved this to the onyx_confluence.py file

File: backend/danswer/connectors/confluence/connector.py [hunk 3]
```diff
@@ -118,21 +116,6 @@ def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None
             is_cloud=self.is_cloud,
             wiki_base=self.wiki_base,
         )
-
```
Comment (hagen-danswer) [inline backend/danswer/connectors/confluence/connector.py:121]: moved this to the onyx_confluence.py file

**Review by rkuo-danswer — APPROVED** [review]

File: backend/danswer/background/celery/tasks/external_group_syncing/tasks.py [hunk 4]
```diff
@@ -29,7 +29,7 @@
 from ee.danswer.db.connector_credential_pair import get_all_auto_sync_cc_pairs
 from ee.danswer.db.external_perm import ExternalUserGroup
 from ee.danswer.db.external_perm import replace_user__ext_group_for_cc_pair
-from ee.danswer.external_permissions.sync_params import EXTERNAL_GROUP_SYNC_PERIOD
+from ee.danswer.external_permissions.sync_params import EXTERNAL_GROUP_SYNC_PERIODS
```
Comment (hagen-danswer) [inline backend/danswer/background/celery/tasks/external_group_syncing/tasks.py:32]: individually control group sync periods based on source

**Pull request merged**